pymbolic/mapper/unifier.py

```diff
@@ -43,9 +43,8 @@ def unify_map(map1, map2):
 
 
 class UnificationRecord(object):
-    def __init__(self, equations, lmap=None, rmap=None):
-        self.equations = equations
 
+    def __init__(self, equations, lmap=None, rmap=None):
         # lmap and rmap just serve as a tool to reject
```

Comment:
You're breaking the interface. So don't pretend like you aren't. Having constructor parameters (`equations`, `rmap`) silently go nowhere is not nice. Just cut them.

Also get rid of the now-incorrect comment.